chromium_src/chrome/browser/lifetime/application_lifetime_desktop.cc

```diff
@@ -0,0 +1,25 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "chrome/browser/lifetime/application_lifetime_desktop.h"
+
+#include "brave/browser/ui/webui/settings/brave_relaunch_handler_mac.h"
+
+#define AttemptRestart AttemptRestart_ChromiumImpl
```

Comment:
what about `RelaunchIgnoreUnloadHandlers()` (that uses then same `AttemptRestartInternal()` internally)?

Reply:
I'm sorry, I am not exactly sure what you are suggesting.

Are you suggesting to replace all occurrences of `AttemptRestart` in this file by `RelaunchIgnoreUnloadHandlers`? If yes, then - without further changes - that latter function simply won't be called when the user clicks a Relaunch button in `brave://settings`.

Or are you suggesting to call `RelaunchIgnoreUnloadHandlers()` from `AttemptRestart()`? If yes, then that will break the contract of `AttemptRestart`.

Finally, what's the motivation for using `RelaunchIgnoreUnloadHandlers` instead of `AttemptRestart` at all? Per the comments in the code, `RelaunchIgnoreUnloadHandlers` "differs from `AttemptRelaunch` in that all user prompts (e.g., beforeunload handlers and confirmation to abort in-progress downloads) are bypassed". Why would we want to bypass those prompts? It seems that they make sense for a Relaunch button in `brave://settings`.

Reply:
I assume that @atuchin-m suggests to override `RelaunchIgnoreUnloadHandlers()` like you did 
 for `AttemptRestart()` as `RelaunchIgnoreUnloadHandlers()` could be called when update is ready.
something like this?
```
void RelaunchIgnoreUnloadHandlers() {
  if (brave_relaunch_handler::RestartOnMac()) {
    return;
  }
  RelaunchIgnoreUnloadHandlers_ChromiumImpl();
}
```

Reply:
My point was that both functions in `application_lifetime_desktop.cc` should be adjusted.

Reply:
👍 . They're both adjusted now.